### PR TITLE
Migrate size constant configuration to plugins.yaml

### PR DESCRIPTION
**Pull request by ashleydavies**
This change moves the `size.go` constants into a config file (it keeps them in the `go` file as a fallback, so it can still run happily with an empty config file)

### CODE REVIEW

File: prow/plugins.yaml [hunk 1]
```diff
@@ -73,6 +73,13 @@ approve:
   implicit_self_approve: true
   lgtm_acts_as_approve: true
 
+size:
+  sLines: 20
+  mLines: 50
+  lLines: 150
+  xlLines: 400
+  xxlLines: 800
```
[the remark this answers is not in the document]
Comment (ashleydavies) [reply prow/plugins.yaml]: My bad - this wasn't intentional; sorry!

**Comment by ashleydavies** [comment]
No problem! Glad to help out; it's proving very useful for us!

File: prow/plugins.yaml [hunk 2]
```diff
@@ -73,6 +73,14 @@ approve:
   implicit_self_approve: true
   lgtm_acts_as_approve: true
 
+# Lower bounds; XS is assumed to be zero.
```
Comment (cblecker) [inline prow/plugins.yaml]: The comment should specify that the value is number of lines (not explicitly clear).

File: prow/plugins/plugins.go [hunk 3]
```diff
@@ -285,6 +286,15 @@ type SigMention struct {
 	Re     *regexp.Regexp `json:"-"`
 }
 
+// Size specifies configuration for the size plugin, allowing configurable lower bounds for each size label
```
Comment (cblecker) [inline prow/plugins/plugins.go]: This comment too.. should specify that these are in number of lines

**Comment by cblecker** [comment]
cc @mattfarina on this. I know this was one of those little things that the helm project wished they could configure. This will soon be possible.

**Comment by stevekuznetsov** [comment]
As written this is global configuration for the plugin, not per-repo

File: prow/plugins/plugins.go [hunk 4]
```diff
@@ -285,6 +286,15 @@ type SigMention struct {
...
+type Size struct {
+	S   int
```
Comment (stevekuznetsov) [inline prow/plugins/plugins.go]: These need `json:"s"` annotations for config loading

**Comment by ashleydavies** [comment]
@cblecker I could add to this to have per-repo config if that's something that would be particularly helpful?

**Comment by stevekuznetsov** [comment]
/lgtm
/hold
@cblecker @cjwagner any final comments?

**Comment by cblecker** [comment]
This is good. I think a future iteration to make this configurable per-repo would be even better, but it shouldn't hold this up.

/lgtm

File: prow/plugins/plugins.go [hunk 5]
```diff
@@ -285,6 +286,15 @@ type SigMention struct {
 	Re     *regexp.Regexp `json:"-"`
 }
 
+// Size specifies configuration for the size plugin, defining lower bounds (in # lines changed) for each size label
```
Comment (cjwagner) [inline prow/plugins/plugins.go]: Mention how XS is determine here as well.

**Comment by ashleydavies** [comment]
Thanks for all the feedback everyone! Happy for this to be merged now if there's no other issues; I'll pop up a PR for per-repo customisation if I find the time but it may be in a few weeks.

File: prow/plugins.yaml [hunk 6]
```diff
@@ -73,6 +73,14 @@ approve:
   implicit_self_approve: true
   lgtm_acts_as_approve: true
 
+# Lower bounds in number of lines changed; XS is assumed to be zero.
```
Comment (BenTheElder) [inline prow/plugins.yaml]: might be worth mentioning that there is an attempt to exclude certain kinds of generated lines 
(which also should perhaps be more configurable...)

**Comment by stevekuznetsov** [comment]
Clarifying generated files changed sounds like a good follow-up

/lgtm
/hold cancel

**Comment by mattfarina** [comment]
@cblecker Thanks for letting me know. I was on vacation so I'm just getting caught up now.

File: prow/plugins.yaml (hunk 6 above)
Comment (ashleydavies) [reply prow/plugins.yaml]: @BenTheElder It's on my radar to add support for looking in `.gitattributes` for vendored/generated directives as it's recognised by GitHub already for identifying generated code (we're currently duplicating the information between `.generated_files` for this plugin and `.gitattributes` for GitHub) -- is that something which would be welcome?

Comment (BenTheElder) [reply prow/plugins.yaml]: I'd guess so, I have no idea who set up the initial "ignore generated files", I've just run into it before when a huge PR got marked size/s :joy:

**Pull request merged**